src/Stack/Coverage.hs

```diff
@@ -244,7 +244,7 @@ generateHpcReportForTargets opts = do
                              case nc of
                                  CTest testName ->
                                      liftM (pkgPath </>) $ parseRelFile (T.unpack testName ++ "/" ++ T.unpack testName ++ ".tix")
-                                 _ -> fail $
+                                 _ -> liftIO . fail $
```

Comment:
It wouldn't likely be better to use `throwString` anyway. But is this caused by a change in `rio`, or a change in GHC?

Reply:
that's now exported `MonadFail` but `throwString` probably makes more sense, thanks